cpp/src/strings/case.cu

```diff
@@ -59,11 +67,34 @@ struct upper_lower_fn
     character_flags_table_type case_flag; // flag to check with on each character
     const character_flags_table_type* d_flags;
     const character_cases_table_type* d_case_table;
+    const character_special_case_hash_indices_table_type *d_special_case_hash_indices;
+    const special_case_mapping *d_special_case_mapping;
     const int32_t* d_offsets{};
     char* d_chars{};
+    
+    // compute-size / copy the bytes representing the special case mapping for this codepoint
+    __device__ int32_t handle_special_case_bytes(uint32_t code_point, char*& d_buffer, detail::character_flags_table_type flag)
+    {
+        special_case_mapping m = d_special_case_mapping[d_special_case_hash_indices[get_special_case_hash_index(code_point)]];
```

Comment:
This still seems like alot. I have a `code_point` and I need a `special_case_mapping` object. Do I really care how many tables are involved and how to use them?
```
special_case_mapping get_special_case_mapping_for_code_point( uint23_t code_point ) {...}
```
Something like that. 
This way we can completely change how the mapping works and don't have come back here update this line.

Reply:
Agree should be a function call.